morphic.js

```diff
@@ -7519,6 +7519,10 @@ MenuMorph.prototype.addLine = function (width) {
     this.items.push([0, width || 1]);
 };
 
+MenuMorph.prototype.addSubmenu = function (key, submenu) {
+	this.addItem(key, key, null, null, null, null, null, submenu);
```

Comment:
>an API that uses parameters/objects instead of just positional arguments

ES6, of course:

```js
function myComplicatedFunction({
  foo = 'default foo',
  bar = 123,
  kaz = 3 * 7
} = {}) {
 console.log(foo, bar, kaz)
}

myComplicatedFunction()
myComplicatedFunction({foo: 75})
```